docs/release.md

```diff
@@ -0,0 +1,43 @@
+# Release
+
+* Update Makefile variable `VERSION` to the appropiate release version:
+  * alpha: `VERSION ?= v1.15.5-alpha.3`, for operator development before doing a final release
+  * stable: `VERSION ?= v1.15.5`, once alpha releases have been tested successfully
+
+**IMPORTANT**: `VERSION` semver, **must coincide with the original helm chart release**,
+because it's used to download the [original helm chart](https://artifacthub.io/packages/helm/banzaicloud-stable/vault-operator)
+by the `download-helm-chart` makefile target.
+
+For **alpha** releases, the only the semver prefix is used to fetch the upstream chart release.
```

Comment:
The `-alpha.X` is also part of the semver spec, so something like this would be better:

```suggestion
For **alpha** releases, the only the part before the `-` is used to fetch the upstream chart release.
```

Just a little bit of nitpicking :P